content/en/docs/concepts/workloads/controllers/replicaset.md

```diff
@@ -88,66 +102,114 @@ Events:
+In this manner, a ReplicaSet can own a non-homogenous set of Pods
 
-The `.spec.template.metadata.labels` must match the `.spec.selector`, or it will
-be rejected by the API.
+## Writing a ReplicaSet Manifest
 
+As with all other Kubernetes API objects, a ReplicaSet needs the `apiVersion`, `kind`, and `metadata` fields.
+For ReplicaSets, the kind is always just ReplicaSet.
 In Kubernetes 1.9 the API version `apps/v1` on the ReplicaSet kind is the current version and is enabled by default. The API version `apps/v1beta2` is deprecated.
+Please refer to the first lines of the `frontend.yaml` example for guidance.
 
-Also you should not normally create any pods whose labels match this selector, either directly, with 
-another ReplicaSet, or with another controller such as a Deployment. If you do so, the ReplicaSet thinks that it 
-created the other pods. Kubernetes does not stop you from doing this.
+A ReplicaSet also needs a [`.spec` section](https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status).
+
+### Pod Template
 
-If you do end up with multiple controllers that have overlapping selectors, you
-will have to manage the deletion yourself.
+The `.spec.template` is a [pod template](/docs/concepts/workloads/pods/pod-overview/#pod-templates) which also requires
+to have labels in place. In our `frontend.yaml` example we had one label: `tier: frontend`.
+Be careful not to overlap with the selectors of other controllers, lest they try to adopt this pod.
 
-### Labels on a ReplicaSet
+For the template's [restart policy](/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy) field,
+`.spec.template.spec.restartPolicy`, the only allowed  value for  is `Always`, which is the default.
 
-The ReplicaSet can itself have labels (`.metadata.labels`).  Typically, you
-would set these the same as the `.spec.template.metadata.labels`.  However, they are allowed to be
-different, and the `.metadata.labels` do not affect the behavior of the ReplicaSet.
+### Pod Selector
+
+The `.spec.selector` field is a [label selector](/docs/concepts/overview/working-with-objects/labels/). As discussed
+[earlier](#how-a-replicaset-works) these are the labels used to identify potential pods to acquire. In our
+`frontend.yaml` example, the selector was:
+```shell
+matchLabels:
+	tier: frontend
+```
+
+The `.spec.template.metadata.labels` must match the `.spec.selector`, or it will
+be rejected by the API.
 
 ### Replicas
 
-You can specify how many pods should run concurrently by setting `.spec.replicas`. The number running at any time may be higher
-or lower, such as if the replicas were just increased or decreased, or if a pod is gracefully
-shut down, and a replacement starts early.
+You can specify how many pods should run concurrently by setting `.spec.replicas`. The ReplicaSet will create/delete
+its pods to match this number.
 
 If you do not specify `.spec.replicas`, then it defaults to 1.
 
```

Comment:
Below: you must set `propagationPolicy` to `Background` or `Foreground` in delete option. e.g. :

Our style is to avoid Latin phrases like e.g.
Suggestion: ...in the `-d` option. For example:

Same applies to the second curl example.